docs/dev/Pagination-v2.md

```diff
@@ -206,55 +651,59 @@ sequenceDiagram
     participant OpenSearchScrollRequest
     participant ContinuePageRequest
 
-PlanSerializer->>+ProjectOperator: getPlanForSerialization
-  ProjectOperator-->>-PlanSerializer: this
-PlanSerializer->>+ProjectOperator: serialize
-  Note over ProjectOperator: dump private fields
-  ProjectOperator->>+ResourceMonitorPlan: getPlanForSerialization
-    ResourceMonitorPlan-->>-ProjectOperator: delegate
-  Note over ResourceMonitorPlan: ResourceMonitorPlan<br />is not serialized
-  ProjectOperator->>+OpenSearchPagedIndexScan: serialize
+PlanSerializer ->>+ ProjectOperator : getPlanForSerialization
+  ProjectOperator -->>- PlanSerializer : this
+PlanSerializer ->>+ ProjectOperator : serialize
+  Note over ProjectOperator : dump private fields
+  ProjectOperator ->>+ ResourceMonitorPlan : getPlanForSerialization
+    ResourceMonitorPlan -->>- ProjectOperator : delegate
+  Note over ResourceMonitorPlan : ResourceMonitorPlan<br />is not serialized
+  ProjectOperator ->>+ OpenSearchPagedIndexScan : serialize
     alt First page
-      OpenSearchPagedIndexScan->>+OpenSearchScrollRequest: toCursor
-        OpenSearchScrollRequest-->>-OpenSearchPagedIndexScan: scroll ID
+      OpenSearchPagedIndexScan ->>+ OpenSearchScrollRequest : toCursor
+        OpenSearchScrollRequest -->>- OpenSearchPagedIndexScan : serialized request
     else Subsequent page
-      OpenSearchPagedIndexScan->>+ContinuePageRequest: toCursor
-        ContinuePageRequest-->>-OpenSearchPagedIndexScan: scroll ID
+      OpenSearchPagedIndexScan ->>+ ContinuePageRequest : toCursor
+        ContinuePageRequest -->>- OpenSearchPagedIndexScan : serialized request
     end
-    Note over OpenSearchPagedIndexScan: dump private fields
-    OpenSearchPagedIndexScan-->>-ProjectOperator: serialized
-  ProjectOperator-->>-PlanSerializer: serialized
-Note over PlanSerializer: Zip to reduce size
+    Note over OpenSearchPagedIndexScan : dump private fields
+    OpenSearchPagedIndexScan -->>- ProjectOperator : serialized
+  ProjectOperator -->>- PlanSerializer : serialized
+Note over PlanSerializer : Zip to reduce size
 ```
 
-### Deserialization
+#### Deserialization
+
+Deserialization restores previously serialized Physical Plan tree. The recovered tree is ready to execute and should return the next page of the search response. To complete the tree restoration, SQL engine should build a new request to the OpenSearch node. This request doesn't contain a search query, but it contains a search context reference -- `scrollID`. To create a new `ContinuePageRequest` object it is require to access to the instance of `OpenSearchStorageEngine`. `OpenSearchStorageEngine` can't be serialized and it exists as a singletone in the SQL plugin engine. `PlanSerializer` creates a customized deserialization binary object stream -- `CursorDeserializationStream`; this stream provides an interface to access to the `OpenSearchStorageEngine` object.
```

Comment:
```suggestion
Deserialization restores previously serialized Physical Plan tree. The recovered tree is ready to execute and should return the next page of the search response. To complete the tree restoration, SQL engine should build a new request to the OpenSearch node. This request doesn't contain a search query, but it contains a search context reference -- `scrollID`. To create a new `ContinuePageRequest` object it is require to access to the instance of `OpenSearchStorageEngine`. `OpenSearchStorageEngine` can't be serialized and it exists as a singleton in the SQL plugin engine. `PlanSerializer` creates a customized deserialization binary object stream -- `CursorDeserializationStream`. This stream provides an interface to access the `OpenSearchStorageEngine` object.
```

Reply:
> This stream provides an interface to access the `OpenSearchStorageEngine` object.
I'm not sure what you mean by this?  Does this mean we can create a new `OpenSearchStorageEngine` object?

Reply:
No, we can't, it is a singleton.